### PR TITLE
Easy require

### PR DESCRIPTION
**Pull request by jasonkarns**
This change makes jshint-stylish easier to require from the jshint CLI.

Before this change, in order to use jshint-stylish with jshint CLI:
```
jshint --reporter node_modules/jshint-stylish/stylish.js
```

With this change, one can supply just the package path and let package.json's `main` take effect:
```
jshint --reporter node_modules/jshint-stylish
```

This change is still compatible with gulp-jshint.

As for grunt-contrib-jshint, as of 0.11.1 the reporter config option is duck-typed to a string. So with this change, it can still be supplied via `require`:
``` js
jshint: {
    options: {
        reporter: require('jshint-stylish')
    },
    target: ['file.js']
}
```

### CODE REVIEW

**Comment by sindresorhus** [comment]
Agh, JSHint. You should just be able to require reporters as normal node modules. ESLint does this nicely I believe. Eg. `jshint --reporter jshint-stylish`. But sure, I guess this is better than nothing.

File: stylish.js [hunk 1]
```diff
@@ -9,6 +9,9 @@ function pluralize(str, count) {
 }
```
Comment (sindresorhus) [inline stylish.js]: Just rename this file to `index.js` now.

File: package.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
   "version": "1.0.1",
   "description": "Stylish reporter for JSHint",
   "license": "MIT",
+  "main": "stylish",
```
Comment (sindresorhus) [inline package.json]: this can be removed because of the other comment

**Comment by jasonkarns** [comment]
done

**Comment by kevva** [comment]
@jasonkarns, can you rebase?

**Comment by jasonkarns** [comment]
rebased and test fixed (lint warnings still being ignore per `master`)

also worth noting, this change was already going to be backwards incompatible with grunt-contrib-jshint pre v0.11.1, so a major version bump is advised.

and now that we've eliminated `stylish.js`, it will be backwards-incompatible for anyone who was using the jshint cli. so definitely a major version bump.

**Pull request closed without merging**